Automattic/jetpack
# Sync: Fix syncing of site icon URL

While testing 4.3, I reset all of my Jetpack options, which also delete the `jetpack_site_icon_url` option. Since then, I've noticed that my site icon hasn't been syncing properly. After looking into the issue, it seems that were trying to set the `jetpack_site_icon_url` option, but that was no longer there for my site.

Since we don't load the site icon module when core's site icon functionality exists, this option will no longer be created on my site. 😞 

So, the solution here is to begin syncing the site icon as a callable. So, we try to use the core function `get_site_icon_url()` to get the site icon, and if there is no core site icon, we fallback to the `jetpack_site_icon_url` option.

To test:

- Create new install of Jetpack or delete all Jetpack options
- Set site icon
- Sync
- Ensure site icon is in site response

cc @lezama @enejb for code review
